browser_history/generic.py

```diff
@@ -167,34 +173,57 @@ def fetch(self, history_paths=None, sort=True, desc=False):
             history_paths = self.history_paths()
         output_object = Outputs()
         with tempfile.TemporaryDirectory() as tmpdirname:
+            length = 0
             for history_path in history_paths:
                 copied_history_path = shutil.copy2(history_path.absolute(), tmpdirname)
                 conn = sqlite3.connect(f"file:{copied_history_path}?mode=ro", uri=True)
                 cursor = conn.cursor()
-                cursor.execute(self.history_SQL)
-                date_histories = [
-                    (
-                        datetime.datetime.strptime(d, "%Y-%m-%d %H:%M:%S").replace(
-                            tzinfo=_local_tz
-                        ),
-                        url,
-                    )
-                    for d, url in cursor.fetchall()
-                ]
-                output_object.entries.extend(date_histories)
+                if type=='history':
+                    cursor.execute(self.history_SQL)
+                    date_histories = [
+                        (
+                            datetime.datetime.strptime(d, "%Y-%m-%d %H:%M:%S").replace(
+                                tzinfo=_local_tz
+                            ),
+                            url,
+                        )
+                        for d, url in cursor.fetchall()
+                    ]
+                    output_object.history_entries.extend(date_histories)
+                    if sort:
+                        output_object.history_entries.sort(reverse=desc)
+                elif type == 'bookmarks':
+                    cursor.execute(self.bookmarks_SQL)
+                    date_bookmarks = [
+                        (
+                            datetime.datetime.strptime(d, "%Y-%m-%d %H:%M:%S").replace(
+                                tzinfo=_local_tz
+                            ),
+                            url,
+                            title,
+                            folder,
+                        )
+                        for d, url,title,folder in cursor.fetchall()
+                    ]
+                    if(len(date_bookmarks)<=length):
```

Comment:
The `length` variable initialized in line 176 is used here. Is the value of `length` being changed anywhere else? If not wouldn't it be better to use the hardcoded value of 0 instead of using an other variable?

I'm saying this in view of the code quality. `pylint` will raise warnings if many variables are used in a function. There's a limit of 9 or something. I'm not sure though.

Reply:
Firefox has two history files for some reason on linux, maybe its to store history without consuming too much space, anyways in both the history files there are redundant copies of bookmarks, one which keeps updating and one stagnant,
that is why i kept this variable, I missed a commit on this i guess, the next line should have been, 
```bash 
length  = len(date_bookmarks)  ```

Reply:
Ah okay. Makes sense. :smile: